core/src/main/resources/defaults/blueprints/cdp-sdx-medium-ha.bp

```diff
@@ -0,0 +1,428 @@
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "kafka-GATEWAY-BASE",
+            "roleType": "GATEWAY"
+          },
+          {
+            "base": true,
+            "refName": "kafka-KAFKA_BROKER-BASE",
+            "roleType": "KAFKA_BROKER"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          },
+          {
+            "name": "offsets.topic.replication.factor",
+            "value": "1"
+          }
+        ],
+        "serviceType": "KAFKA"
+      },
+      {
+        "refName": "ranger",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "ranger-RANGER_USERSYNC-BASE",
+            "roleType": "RANGER_USERSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_TAGSYNC-BASE",
+            "roleType": "RANGER_TAGSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_ADMIN-BASE",
+            "roleType": "RANGER_ADMIN"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "rangeradmin_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "rangertagsync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "solr_service",
+            "ref": "solr"
+          },
+          {
+            "name": "rangerusersync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "keyadmin_user_password",
+            "value": "{{{ general.password }}}"
+          }
+        ],
+        "serviceType": "RANGER"
+      },
+      {
+        "refName": "solr",
+        "roleConfigGroups": [
+          {
+            "refName": "solr-SOLR_SERVER-BASE",
+            "roleType": "SOLR_SERVER",
+            "base": true
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          }
+        ],
+        "serviceType": "SOLR"
+      },
+      {
+        "refName": "zookeeper",
+        "serviceType": "ZOOKEEPER",
+        "roleConfigGroups": [
+          {
+            "refName": "zookeeper-SERVER-BASE",
+            "roleType": "SERVER",
+            "base": true
+          }
+        ]
+      },
+      {
+        "serviceType": "KNOX",
+        "refName": "knox",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "knox-KNOX_GATEWAY-BASE",
+            "roleType": "KNOX_GATEWAY",
+            "configs": [
+              {
+                "name": "gateway_master_secret",
```

Comment:
Is this not supposed to be injected from Java code?

Reply:
Currently, it is not. "gateway_master_secret" is a mandatory config which is not injected. Will create a Jira for it. we can remove once that capability is added.

Reply:
https://github.com/hortonworks/cloudbreak/commit/70c0b6788fe3a5d3ebe555d50c46d39848e0c4c4. It's already on master

Reply:
No need for the Jira, it was already implemented in 70c0b6788fe3a5d3ebe555d50c46d39848e0c4c4, but the version with the change may not be available yet in all environments.

Reply:
created jira for this config to be injected by CB.

Reply:
Sorry guys, I missed your responses and went ahead with the jira's. I will close them.